### PR TITLE
[stable-v2.8] Merge avs-tplg and sof-ace-tplg under production directory 

### PR DESCRIPTION
**Pull request by marc-hb**
This is a backport to stable-v2.8 of a subset of mega commit 65e4c1f8b519 ("topology2: Merge avs-tplg and sof-ace-tplg under production directory") on the main branch.

This is the absolute minimum subset that meets CI's new topology expectations and allows to permanently run all tests on stable-v2.8 going forward.

Unlike that mega commit, this PR is not just smaller as a whole but also split into much smaller, separate commits that can be easily reviewed for correctness and to make sure it does exactly what its commit message says. Each commit builds and it's easy to compare build directories from one commit to the next and check that topology files are only moved around without any change (and I actually did that).

### CODE REVIEW

**Comment by marc-hb** [comment]
https://sof-ci.01.org/sofpr/PR8861/build2705/devicetest/index.html is looking good so far.

There was some DUT-specific issue with https://sof-ci.01.org/sofpr/PR8861/build2704/devicetest/index.html (internal run 37897) so I aborted it.

**Comment by marc-hb** [comment]
SOFCI TEST

**Comment by marc-hb** [comment]
Only one multiple pause/resume failure in ACE https://sof-ci.01.org/sofpr/PR8861/build2707/devicetest/index.html after 30 minutes. Maybe related to #8770? Everything else is green.

Only one suspend/resume failure in CAVS https://sof-ci.01.org/sofpr/PR8861/build2708/devicetest/index.html

QB cannot only test the main branch, known issue: https://sof-ci.01.org/sof-pr-viewer/#/build/PR8861/build13564131

File: tools/topology/topology2/production/tplg-targets-hda-generic.cmake [hunk 1]
```diff
@@ -6,4 +6,13 @@ DEEPBUFFER_D0I3_COMPATIBLE=true"
 # HDA topology with mixer-based pipelines for HDA and passthrough pipelines for HDMI
 "sof-hda-generic\;sof-hda-generic\;HDA_CONFIG=mix,DEEPBUFFER_FW_DMA_MS=100,\
 DEEPBUFFER_D0I3_COMPATIBLE=true"
+
+# HDA topology with mixer-based pipelines for HDA and
+# passthrough pipelines for HDMI and
+# 2 or 4 DMIC, no NHLT blob included in topology
+"sof-hda-generic\;sof-hda-generic-2ch\;HDA_CONFIG=mix,NUM_DMICS=2,\
+DEEPBUFFER_FW_DMA_MS=100,DEEPBUFFER_D0I3_COMPATIBLE=true"
+"sof-hda-generic\;sof-hda-generic-4ch\;HDA_CONFIG=mix,NUM_DMICS=4,\
```
Comment (ranj063) [inline tools/topology/topology2/production/tplg-targets-hda-generic.cmake:15]: [8:49 PM] Sridharan, Ranjani
when you have NUM_DMICS as 4 you also need PDM1_MIC_A_ENABLE=1,PDM1_MIC_B_ENABLE=1

Comment (marc-hb) [reply tools/topology/topology2/production/tplg-targets-hda-generic.cmake:15]: Please correct me but I think it is there... I copied it from 65e4c1f8b5192f0ee

**Review by kv2019i — APPROVED** [review]
Looks good. So this will drop the NHLT blobs from the generic topologies (to implement https://github.com/thesofproject/sof/issues/8683), but as use of these blobs is behind a kernel module parameter that is off by default, this will not affect any default configuration.

**Review by lgirdwood — APPROVED** [review]

File: tools/topology/topology2/avs-tplg/tplg-targets.cmake [hunk 2]
```diff
@@ -1,18 +1,18 @@
 # SPDX-License-Identifier: BSD-3-Clause
 
 # Array of "input-file-name;output-file-name;comma separated pre-processor variables"
-set(TPLGS
+list(APPEND TPLGS
```
Comment (ujfalusi) [inline tools/topology/topology2/avs-tplg/tplg-targets.cmake:4]: This change is not related to topic, unless it is part of the 'preparation' umbrella.

File: tools/topology/topology2/avs-tplg/tplg-targets.cmake [hunk 3]
```diff
@@ -2,10 +2,7 @@
 
 # Array of "input-file-name;output-file-name;comma separated pre-processor variables"
 list(APPEND TPLGS
-# CAVS HDMI only topology with passthrough pipelines
-"sof-hda-generic\;sof-hda-generic-idisp\;DEEPBUFFER_FW_DMA_MS=100"
-# CAVS HDA topology with mixer-based pipelines for HDA and passthrough pipelines for HDMI
-"sof-hda-generic\;sof-hda-generic\;HDA_CONFIG=mix,DEEPBUFFER_FW_DMA_MS=100"
```
Comment (ujfalusi) [inline tools/topology/topology2/avs-tplg/tplg-targets.cmake:8]: why?

File: tools/topology/topology2/production/tplg-targets-ace.cmake [hunk 4]
```diff
@@ -2,12 +2,6 @@
 
 # Array of "input-file-name;output-file-name;comma separated pre-processor variables"
 list(APPEND TPLGS
-# HDMI only topology with passthrough pipelines
-"sof-hda-generic\;sof-hda-generic-idisp\;DEEPBUFFER_FW_DMA_MS=100,\
-DEEPBUFFER_D0I3_COMPATIBLE=true"
-# HDA topology with mixer-based pipelines for HDA and passthrough pipelines for HDMI
-"sof-hda-generic\;sof-hda-generic\;HDA_CONFIG=mix,DEEPBUFFER_FW_DMA_MS=100,\
-DEEPBUFFER_D0I3_COMPATIBLE=true"
```
Comment (ujfalusi) [inline tools/topology/topology2/production/tplg-targets-ace.cmake:10]: could have been done in the previous patch..

File: tools/topology/topology2/production/tplg-targets-hda-generic.cmake [hunk 5]
```diff
@@ -6,4 +6,13 @@ DEEPBUFFER_D0I3_COMPATIBLE=true"
 # HDA topology with mixer-based pipelines for HDA and passthrough pipelines for HDMI
 "sof-hda-generic\;sof-hda-generic\;HDA_CONFIG=mix,DEEPBUFFER_FW_DMA_MS=100,\
 DEEPBUFFER_D0I3_COMPATIBLE=true"
+
+# HDA topology with mixer-based pipelines for HDA and
+# passthrough pipelines for HDMI and
+# 2 or 4 DMIC, no NHLT blob included in topology
+"sof-hda-generic\;sof-hda-generic-2ch\;HDA_CONFIG=mix,NUM_DMICS=2,\
```
Comment (ujfalusi) [inline tools/topology/topology2/production/tplg-targets-hda-generic.cmake:13]: so, at the previous patch the tplg2 build was incomplete?
I know bisecting is not a primary concern, just pointing out.

**Review by ujfalusi — COMMENTED** [review]
@marc-hb, I'm not sure what benefit this specific split brings, patch three is still - as you called it - a mega patch...
If you must split, I would recommend:
- avs-tplg: introduce the set of hda-generic tplgs w/ and w/o NHLT
- sof-ace-tplg: introduce the set of hda-generic tplgs w/ and w/o NHLT
- mega patch to merge

imho a single revert to get back to the old state is cleaner in some ways when doing such intrusive change anyways.

**Review by ujfalusi — APPROVED** [review]

**Comment by marc-hb** [comment]
> so, at the previous patch the tplg2 build was incomplete?

Possibly yes.

> I know bisecting is not a primary concern, just pointing out.

It depends what you mean by "bisecting". If bisecting implies running tests then no, it's not a concern inside this PR. Running tests in CI is not possible before this PR anyway (and obviously possible after that **complete** PR - that's the entire purpose).

On the other hand, I did "bisect" this PR by carefully comparing the build directories before versus after each commit. Each commit is crafted to make _that_ sort of "bisect" easier. For the same reason, each commit is focused on one change and one change only to make it easier to review.

File: tools/topology/topology2/avs-tplg/tplg-targets.cmake (hunk 2 above)
Comment (marc-hb) [reply tools/topology/topology2/avs-tplg/tplg-targets.cmake:4]: Correct, this could  have been another "preparation" commit. However I felt like this commit was small and easy enough to review. Moreover this CMake change makes no output difference at this early stage.

File: tools/topology/topology2/avs-tplg/tplg-targets.cmake (hunk 3 above)
Comment (marc-hb) [reply tools/topology/topology2/avs-tplg/tplg-targets.cmake:8]: Otherwise these would clash in the next commit with the identical ones in sof-ace-tplg/

**Comment by marc-hb** [comment]
>  patch three is still - as you called it - a mega patch...

I respectfully disagree. Patch 3 has two "pure" renames + small CMake changes + 1 delete CMakeLists.txt
